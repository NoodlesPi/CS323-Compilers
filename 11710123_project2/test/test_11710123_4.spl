int helloworld(int a){
    return a;
}
// type 8
int helloworld(float b){
    return b;
}
int helloworld(int c, int d){
    return (c + d)/2;
}
// type 7 and type 4
int helloworld(int a){
    return a / 'c';
}
int main(){
    // type 9
    helloworld(helloworld(1), helloworld(2), helloworld(helloworld(1), helloworld(2)), helloworld(helloworld(1), helloworld(2), helloworld(3)));
    return 1.2;
}