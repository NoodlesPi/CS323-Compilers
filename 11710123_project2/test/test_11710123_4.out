Error type 8 at Line 6: the function’s return value type mismatches the declared type
Error type 7 at Line 13: unmatching operands, such as adding an integer to a structure variable
Error type 8 at Line 13: the function’s return value type mismatches the declared type
Error type 4 at Line 12: function is redefined
Error type 9 at Line 16: the function’s arguments mismatch the declared parameters
Error type 9 at Line 16: the function’s arguments mismatch the declared parameters
Error type 8 at Line 17: the function’s return value type mismatches the declared type
